File: test/java/hashing/TwoSumTest.java
package java.hashing;

import org.junit.jupiter.api.Test;

import static java.hashing.TwoSum.twoSum;
import static org.junit.jupiter.api.Assertions.assertArrayEquals;

class TwoSumTest {

    @Test
    void twoSumTestPositive() {
        assertArrayEquals(new int[]{0, 4},
                twoSum(new int[]{5, 2, 7, 10, 3, 9}, 8),
                "Two sum positive test");
    }

    @Test
    void twoSumTestNegative() {
        assertArrayEquals(new int[]{-1, -1},
                twoSum(new int[]{5, 2, 7, 10, 7, 9}, 8),
                "Two sum negative test");
    }
}

File: src/test/kotlin/hashing/TwoSumTest.kt
package hashing

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class TwoSumTest {
    @Test
    fun twoSumPositiveTest() {
        assertArrayEquals(intArrayOf(0, 4),
            TwoSum().twoSum(intArrayOf(5, 2, 7, 10, 3, 9), 8))
    }

    @Test
    fun twoSumNegativeTest() {
        assertArrayEquals(intArrayOf(-1, -1),
            TwoSum().twoSum(intArrayOf(5, 2, 7, 10, 7, 9), 8))
    }
}